Add beta release process (#898)
meilisearch/instant-meilisearch: CONTRIBUTING.md

export const PACKAGE_VERSION = 'X.X.X'
```

> #### Github Publish

Reviewer: ```suggestion
#### GitHub publish
```

Once the changes are merged on `main`, you can publish the current draft release via the [GitHub interface](https://github.com/meilisearch/instant-meilisearch/releases): on this page, click on `Edit` (related to the draft release) > update the description (be sure you apply [these recommendations](https://github.com/meilisearch/integration-guides/blob/main/resources/integration-release.md#writting-the-release-description)) > when you are ready, click on `Publish release`.

GitHub Actions will be triggered and push the package to [npm](https://www.npmjs.com/package/@meilisearch/instant-meilisearch).

#### Codesandbox update
Once the version is available on npm, please update the instant-meilisearch version used in the different Code-Sandboxes we provide:

- [Meilisearch + InstantSearch](https://codesandbox.io/s/ms-is-mese9)
- [Meilisearch + Vue 2 InstantSearch](https://codesandbox.io/s/ms-vue-is-1d6bi)

> - [Meilisearch + Vue 3 InstantSearch](https://codesandbox.io/s/ms-vue3-is-0293zk)

Author: Should not be on this PR, if you want me to open a separate PR with that change, feel free to tell me. Otherwise, if you don't see it I don't see it 🙈

Reviewer: I have not seen anything

GitHub Actions will be triggered and push the package to [npm](https://www.npmjs.com/package/@meilisearch/instant-meilisearch).

#### Codesandbox update
Once the version is available on npm, please update the instant-meilisearch version used in the different Code-Sandboxes we provide:

- [Meilisearch + InstantSearch](https://codesandbox.io/s/ms-is-mese9)
- [Meilisearch + Vue 2 InstantSearch](https://codesandbox.io/s/ms-vue-is-1d6bi)
- [Meilisearch + Vue 3 InstantSearch](https://codesandbox.io/s/ms-vue3-is-0293zk)
- [Meilisearch + React InstantSearch](https://codesandbox.io/s/ms-react-is-sh9ud)

If you don't have the access to do it, please request it internally.

#### Release a `beta` Version

This package is able to create multiple types of betas:

> - A standard package beta, working on the latest version of Meilisearch.

Reviewer: ```suggestion
- A standard package beta, so a instant-meilisearch beta, compatible with the latest version of Meilisearch.
```

Author: > A standard npm package beta, so an Instant-Meilisearch beta, compatible with the latest version of Meilisearch.

These are all going to be published on `npm` with the beta tag. Do you think we should continue referring to instant-meilisearch as `instant-meilisearch` and not `Instant-Meilisearch` as it is the name of the took?

Reviewer: > Unfortunately, these are all going to be published on npm with the beta tag :/

Oh yes you are right for the npm

> Do you think we should continue referring to instant-meilisearch as instant-meilisearch and not Instant-Meilisearch as it is the name of the took?

At your convenience! Sorry I don't remember the common guidelines for it

Here are the steps to release a beta version of this package:

1. Create a new branch containing the changes with the correct name format following these rules:
    - `package beta`: create a branch `beta/xx-xx` with the context of your beta.
      Example: `beta/refactor`.
    - Meilisearch `pre-release beta`: create a branch originating from `bump-meilisearch-v*.*.*` named `pre-release-beta/v*.*.*`. <br>
      Example: `pre-release-beta/v0.30.0`

>     - Meilisearch `protype beta`: create a branch `protoype-beta/xx-xx`. Where `xxx` has the same name as the docker image containing the prototype.

Reviewer: ```suggestion
    - Meilisearch `prototype beta`: create a branch `prototype-beta/xx-xx`. Where `xxx` has the same name as the docker image containing the prototype.
```

>         Example: If the [docker image](https://hub.docker.com/r/getmeili/meilisearch/tags) is named: `v0.29.0-pagination.beta.2`, the should be named: `prototype-beta/pagination`

Reviewer: ```suggestion
        Example: If the [docker image](https://hub.docker.com/r/getmeili/meilisearch/tags) is named: `v0.29.0-pagination.beta.2`, this should be named: `prototype-beta/pagination`
```
